### PR TITLE
Upgrade Shortcodes to 1.0.0

### LINKED ISSUE #7986: Shortcodes 1.0.0
Update Shortcodes to version 1.0.0:
https://www.nuget.org/packages/Shortcodes

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #7986 #7715

### CODE REVIEW

**Comment by agriffard** [comment]
Not so simple, the tests will fail.

**Comment by hishamco** [comment]
> Not so simple, the tests will fail.

No worry I will fix them

**Comment by hishamco** [comment]
Seems the only test that fails is locale shortcode, so I need to figure out if it's a shortcodes bug after the recent commits, because it was working before

**Comment by hishamco** [comment]
Everything is working fine in Shortcodes, the issue occurs from `LocaleShortcodeProvider.Null`, problem solved

**Comment by Skrypt** [comment]
hmm yeah Null is probably a reserved word.

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Providers/LocaleShortcodeProvider.cs [hunk 1]
```diff
@@ -9,19 +9,19 @@ public class LocaleShortcodeProvider : IShortcodeProvider
...
 
-        private static readonly ValueTask<string> Null = new ValueTask<string>((string)null);
+        private static readonly ValueTask<string> Empty = new ValueTask<string>(String.Empty);
 
         public ValueTask<string> EvaluateAsync(string identifier, Arguments arguments, string content, Context context)
         {
             if (identifier != ShortCodeIdentifier)
             {
-                return Null;
+                return Empty;
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Providers/LocaleShortcodeProvider.cs:18]: This should return Null as before

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Providers/LocaleShortcodeProvider.cs [hunk 2]
```diff
@@ -46,7 +46,7 @@ public ValueTask<string> EvaluateAsync(string identifier, Arguments arguments, s
                 }
             }
 
-            return Null;
+            return Empty;
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.Shortcodes/Providers/LocaleShortcodeProvider.cs:49]: This should return Empty

**Review by deanmarcussen — COMMENTED** [review]
The change in behaviour with shortcodes is to render the `[blah]` if it is not registered.

I assume (hope) that means somewhere it's evaluated against null for not registered, and empty for is registered, but didn't return a value.

Can you try with the suggested change to see if it passes, and if not we might have to look at shortcodes.

**Comment by deanmarcussen** [comment]
Fixes https://github.com/OrchardCMS/OrchardCore/issues/7715

File: src/OrchardCore.Modules/OrchardCore.Shortcodes/Providers/LocaleShortcodeProvider.cs (hunk 1 above)
Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Shortcodes/Providers/LocaleShortcodeProvider.cs:18]: Dean if you look before my last commit is was `Null` the issue comes from `null` value

Comment (deanmarcussen) [reply src/OrchardCore.Modules/OrchardCore.Shortcodes/Providers/LocaleShortcodeProvider.cs:18]: I checked shortcodes. This is wrong.

Please apply the suggestions, return Null initially, and Empty if it has no value.

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Shortcodes/Providers/LocaleShortcodeProvider.cs:18]: Let me try your suggestions

**Review by deanmarcussen — APPROVED** [review]

**Pull request merged**